### PR TITLE
Fix ban appeal decider model mixup

### PR DESCRIPTION
**Pull request by freddyheppell**
## Overview of Changes
What this PR was supposed to do:
* Fixes bug where the minecraft player ID of the user that decided on a ban was treated as their account id

The mysterious rabbit hole it ended up going down:
* Migrate to using mariadb for tests since sqlite doesn't support dropping foriegn keys
  * This is in the form of an additional mariadb container, since it is an absolute pain to create a second table in the existing one
* Fixed a bunch of tests that used made up IDs
* Replaced the method for temporarily setting config values with a better and cooler one (since my mocking method broke accessing the db config)
  * And refactored existing tests to use it
* Removed a state on the server factory and made it default, since it's required anyway 

## Deployment Requirements
- [ ] Composer update
- [ ] NPM update
- [ ] Database migration
- [ ] .env update
- [ ] Environment upgrade (eg. PHP version)
- [ ] Other (please specify)

### CODE REVIEW

File: tests/Support/TemporaryConfig.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * Initialise the store of original config values
+     * Run automatically by the base test case
+     *
+     * @return void
+     *
+     * @throws RuntimeException if temporary values were not properly restored last time
+     */
+    public function setUpTemporaryConfig(): void
```
Comment (andyksaw) [inline tests/Support/TemporaryConfig.php:23]: [q] Does this get called by somewhere automatically? I can't find any references to it anywhere

**Review by andyksaw — APPROVED** [review]
Just left one question, but otherwise LGTM! 👍

**Comment by andyksaw** [comment]
Also I'm really surprised how fast those tests ran without the SQLite in-memory database

File: tests/Support/TemporaryConfig.php (hunk 1 above)
Comment (freddyheppell) [reply tests/Support/TemporaryConfig.php:23]: The base test case class [automatically calls](https://github.com/laravel/framework/blob/8235940e7af65affe9a17e2078401d06da4fffd8/src/Illuminate/Foundation/Testing/TestCase.php#L146) `setUpTraitName` and `tearDownTraitName` if they exist on each trait on the test

**Comment by freddyheppell** [comment]
> Also I'm really surprised how fast those tests ran without the SQLite in-memory database

I suspect this is because if you don't use a real DB for testing, it makes all changes in a transaction which is very cheap to discard. Resetting the SQLite DB probably isn't that much faster, because whilst an in-memory DB is quicker to write to, it has to remigrate each time.

**Pull request merged**